ros2_ws/Dockerfile

```diff
@@ -1,15 +1,23 @@
 ARG BASE_TAG=humble
-FROM ros:${BASE_TAG} as base-dependencies
+FROM ros:${BASE_TAG} as environment-variables
 ENV DEBIAN_FRONTEND=noninteractive
 ENV PYTHONWARNINGS=ignore:::setuptools.command.install,ignore:::setuptools.command.easy_install,ignore:::pkg_resources
+ENV PIP_NO_CACHE_DIR 1
```

Comment:
Could you define those in a config file and load it in one line? Would that help or actually not even change anything?

Reply:
I don't think there's a clean way to do it at build time. On build, you can pretty much only set build-args externally.  We can COPY and `RUN source / export` from a file, but that variable will not be available to subsequent layers (unlike the ENV directive).

For docker run, there is an `--env-file` parameter. But that doesn't help us in the CI.

For the second part of the question, I think because ENV directives do not make any changes to the filesystem, they are treated differently as intermediate layers. So, I don't think it's too problematic to have them listed here.